crates/subspace-networking/src/utils.rs

```diff
@@ -52,32 +57,94 @@ impl SubspaceMetrics {
     }
 }
 
-/// Joins async join handle on drop
-pub(crate) struct AsyncJoinOnDrop<T>(Option<Fuse<task::JoinHandle<T>>>);
+/// Joins async join handle on drop.
+/// This future is fused, and will return `Poll::Pending` if polled after completion.
+#[derive(Debug)]
+pub struct AsyncJoinOnDrop<T> {
+    handle: Option<task::JoinHandle<T>>,
+    abort_on_drop: bool,
+}
```

Comment:
This PR manually implements `Fuse`, because wrapping the `JoinHandle` in the `Fuse` type makes `abort()` inaccessible.